### PR TITLE
[now-go] Fix `analyze.go` fails to parse source file with comments

### LINKED ISSUE #2906: [now-go] Fails to parse source file with comments
This function fails to build:

```go
// Package handler (directory named 'api/') is recognized
// by ZEIT Now v2 Lambda platform.
package handler

import (
	"net/http"
	"log"
)

func init() { log.SetFlags(log.Lmicroseconds) }

var _ http.HandlerFunc = Probe // type check

func Probe(w http.ResponseWriter, req *http.Request) {
}
```

Fails with the following error:

```
Parsing AST for "api/probe.go"
Could not find an exported function in "api/probe.go"
```

cc @sophearak

### PR DESCRIPTION
**Pull request by thasophearak**
This PR fixes #2906
Test fixture also included.

### CODE REVIEW

File: packages/now-go/util/analyze.go [hunk 1]
```diff
@@ -209,6 +209,19 @@ func main() {
 					fmt.Print(string(analyzedJSON))
 					os.Exit(0)
 				}
+
+				// fallback, when ast coudn't parse, with multi-line comments
+				paramStr := fmt.Sprintf("%s", param.Type)
+				if strings.Contains(string(paramStr), "http ResponseWriter") {
```
[the remark this answers is not in the document]
Comment (thasophearak) [reply packages/now-go/util/analyze.go]: The above is more precise in term of checking type of params.

And I tried to check both, but `*http.Request` just gave me back pointer value instead `0xc0000cc4a0`, that why i'm go for this.

I think, we should rely on the the first condition. this is good only for fallback, when the above condition not work (edge case).

[the remark this answers is not in the document]
Comment (styfle) [reply packages/now-go/util/analyze.go]: @sophearak I believe the following source file will not work as expected.

Can you add a test for this behavior?

```go
package handler

import (
	"fmt"
	"net/http"
)

func Index1() {
	fmt.Fprintf(w, "one:RANDOMNESS_PLACEHOLDER")
}

func Index2(w http.ResponseWriter) {
	fmt.Fprintf(w, "two:RANDOMNESS_PLACEHOLDER")
}

func Index3(w http.ResponseWriter, req *http.Request) {
	fmt.Fprintf(w, "three:RANDOMNESS_PLACEHOLDER")
}
```

I would expect the output to be `three:RANDOMNESS_PLACEHOLDER`.

My guess is you will need to iterate over the functions twice. The first pass you should check for the exact match and the second pass can check for the fallback.

[the remark this answers is not in the document]
Comment (thasophearak) [reply packages/now-go/util/analyze.go]: @styfle Nice catch, I added more test for this.

However I remove `func Index1()`, since it will cause the build fail, due to syntax error. 
```
func Index1() {
	fmt.Fprintf(w, "one:RANDOMNESS_PLACEHOLDER")
}
```

File: packages/now-go/util/analyze.go [hunk 2]
```diff
@@ -195,11 +194,36 @@ func main() {
 			continue
 		}
 		if fn.Name.IsExported() == true {
-			// find a valid `net/http` handler function
+			// find a valid `http.HandlerFunc` handler function
+			params := rf[fn.Type.Params.Pos()-offset : fn.Type.Params.End()-offset]
+			validHandlerFunc := (strings.Contains(string(params), "http.ResponseWriter") &&
+				strings.Contains(string(params), "*http.Request"))
```
Comment (styfle) [inline packages/now-go/util/analyze.go]: Previously this was checking for `*http.Request http.ResponseWriter`. Why was it changed? Do you want to be able to match 3 parameters?

Comment (thasophearak) [reply packages/now-go/util/analyze.go]: It was bug that I found, it wasn't check as we intended it to be.

**Review by styfle — APPROVED** [review]
Looks good, thanks!
